types/chain_id_test.go

```diff
@@ -37,22 +37,22 @@ func TestParseChainID(t *testing.T) {
 			"invalid chain-id, undefined identifier", "_1-1", true, nil,
 		},
 		{
-			"invalid chain-id, uppercases", "ETHERMINT_1-1", true, nil,
+			"valid chain-id, uppercases", "ETHERMINT_1-1", false, big.NewInt(1),
 		},
 		{
-			"invalid chain-id, mixed cases", "Ethermint_1-1", true, nil,
+			"valid chain-id, mixed cases", "Ethermint_1-1", false, big.NewInt(1),
 		},
 		{
-			"invalid chain-id, special chars", "$&*#!_1-1", true, nil,
+			"valid chain-id, special chars", "$&*#!_1-1", false, big.NewInt(1),
 		},
 		{
-			"invalid eip155 chain-id, cannot start with 0", "ethermint_001-1", true, nil,
+			"valid eip155 chain-id, start with 0", "ethermint_001-1", false, big.NewInt(1),
 		},
 		{
 			"invalid eip155 chain-id, cannot invalid base", "ethermint_0x212-1", true, nil,
 		},
 		{
-			"invalid eip155 chain-id, non-integer", "ethermint_ethermint_9000-1", true, nil,
+			"valid eip155 chain-id, underscore in identifier", "ethermint_ethermint_9000-1", false, big.NewInt(9000),
```

Comment:
how about `ethermint_ethermint_9000-1-1`

based on
```go
"invalid chain-id, double dash", "aragonchain-1-1", true, nil,
// and
"invalid chain-id, double underscore", "aragonchain_1_1", true, nil,
```
```suggestion
			"valid eip155 chain-id, underscore in identifier", "ethermint_ethermint_9000-1", false, big.NewInt(9000),
		},
		{
			"invalid eip155 chain-id, double underscore and double dash in identifier", "ethermint_ethermint_9000-1-1", true, nil,
```

Reply:
reverted